### DISCUSSION TITLE
how to change the final target

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8783] [General]
how to change the other target when hover over a target.

![image](https://user-images.githubusercontent.com/41353266/225188327-5425f8f8-24ef-4df4-b9d7-7e0975349350.png)

so i wanna change the final target.

**Comment by wuxingping** [comment]
> how to change the other target when hover over a target.
> 
> ![image](https://user-images.githubusercontent.com/41353266/225188327-5425f8f8-24ef-4df4-b9d7-7e0975349350.png)
> 
> so i wanna change the final target.

now i know the way to solved the question,i can overwrite the method of  _searchPossibleTargets .but i think this way is not well.
step:
at first,make a connection between text and rect.for example:text._container = rect.
second,add some code  into the method of _searchPossibleTargets.like this: if (target  === 'text') {target = target._container}

`  _searchPossibleTargets(objects, pointer) {
    // Cache all targets where their bounding box contains point.
    let target,
      i = objects.length;
    // Do not check for currently grouped objects, since we check the parent group itself.
    // until we call this function specifically to search inside the activeGroup
    while (i--) {
      const objToCheck = objects[i];
      if (this._checkTarget(objToCheck, pointer)) {
        target = objToCheck;
//     if (target  === 'text') {
//         target = target._container
//       }
        if (isCollection(target) && target.subTargetCheck) {
          const subTarget = this._searchPossibleTargets(target._objects, pointer);
          subTarget && this.targets.push(subTarget);
        }
        break;
      }
    }
    return target;
  }`

**Reply by ShaMan123** [reply]
this is a legit solution I think

**Comment by ShaMan123** [comment]
turn the text into a control

**Reply by wuxingping** [reply]
> turn the text into a control
I tried to do the following,but it only appears when mousedown,I want it to appear when the group is rendered,can you give me some advice?

  tl: new Control({
    x: -0.5,
    y: -0.5,
    offsetY: 16,
    cursorStyle: 'pointer',
    cornerSize:24,
    render: renderIcon,
  }),

function renderIcon(ctx, left, top, styleOverride, fabricObject) {
  var deleteIcon = "data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3C!DOCTYPE svg PUBLIC '-//W3C//DTD SVG 1.1//EN' 'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3E%3Csvg version='1.1' id='Ebene_1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' width='595.275px' height='595.275px' viewBox='200 215 230 470' xml:space='preserve'%3E%3Ccircle style='fill:%23F44336;' cx='299.76' cy='439.067' r='218.516'/%3E%3Cg%3E%3Crect x='267.162' y='307.978' transform='matrix(0.7071 -0.7071 0.7071 0.7071 -222.6202 340.6915)' style='fill:white;' width='65.545' height='262.18'/%3E%3Crect x='266.988' y='308.153' transform='matrix(0.7071 0.7071 -0.7071 0.7071 398.3889 -83.3116)' style='fill:white;' width='65.544' height='262.179'/%3E%3C/g%3E%3C/svg%3E";
  var img = document.createElement('img');
  img.src = deleteIcon;
  var size = this.cornerSize;
  ctx.save();
  ctx.translate(left, top);
  ctx.rotate(util.degreesToRadians(fabricObject.angle));
  ctx.drawImage(img, -size/2, -size/2, size, size);
  ctx.restore();
}

**Reply by ShaMan123** [reply]
repro
